packages/manager/src/features/CloudPulse/Alerts/ContextualView/AlertInformationActionTable.tsx

```diff
@@ -60,115 +66,139 @@ export interface TableColumnHeader {
   label: string;
 }
 
-export interface AlertStatusMap {
-  /**
-   * Alert id to boolean mapping to hold whether that alert id is enabled or not
-   */
-  [alertId: number]: boolean;
-}
-
-// generate the mapping of id to whether entity present in entity_ids list or not for quick access in table row toggles
-const generateStatusMap = (
-  alerts: Alert[],
-  entityId: string
-): AlertStatusMap => {
-  return alerts.reduce<AlertStatusMap>(
-    (previousValue, alert) => ({
-      ...previousValue,
-      [alert.id]: alert.entity_ids.includes(entityId),
-    }),
-    {}
-  );
-};
-
 export const AlertInformationActionTable = (
   props: AlertInformationActionTableProps
 ) => {
-  const { alerts, columns, entityId, error, orderByColumn } = props;
-  const [alertStatusMap, setAlertStatusMap] = React.useState<AlertStatusMap>(
-    generateStatusMap(alerts, entityId)
-  );
+  const { alerts, columns, entityId, entityName, error, orderByColumn } = props;
 
   const _error = error
     ? getAPIErrorOrDefault(error, 'Error while fetching the alerts')
     : undefined;
+  const { enqueueSnackbar } = useSnackbar();
+  const [selectedAlert, setSelectedAlert] = React.useState<Alert>({} as Alert);
+  const [isDialogOpen, setIsDialogOpen] = React.useState<boolean>(false);
+  const [isLoading, setIsLoading] = React.useState<boolean>(false);
+
+  const { mutateAsync: addEntity } = useAddEntityToAlert();
+
+  const { mutateAsync: removeEntity } = useRemoveEntityFromAlert();
 
-  const handleToggle = ({ id }: Alert) => {
-    setAlertStatusMap((previousValue) => {
-      return {
-        ...previousValue,
-        [id]: !previousValue[id],
+  const handleCancel = () => {
+    setIsDialogOpen(false);
+  };
+  const handleConfirm = React.useCallback(
+    (alert: Alert, currentStatus: boolean) => {
+      const payLoad: EntityAlertUpdatePayload = {
```

Comment:
```suggestion
      const payload: EntityAlertUpdatePayload = {
```

Nit: payload is one word, not two, and would not be camel-cased, following convention (also goes for [L97](https://github.com/linode/manager/pull/11785/files#diff-bc47b725d397f57422f2fc06c76663ccb2dc5bfd668b7ada160a6a377497b0caR97))